sources/imdsclient/src/lib.rs

```diff
@@ -65,57 +46,74 @@ impl ImdsClient {
     }
 
     /// Gets `user-data` from IMDS. The user-data may be either a UTF-8 string or compressed bytes.
-    pub async fn fetch_userdata(&mut self) -> Result<Vec<u8>> {
+    pub async fn fetch_userdata(&mut self) -> Result<Option<Vec<u8>>> {
         self.fetch_imds(PINNED_SCHEMA, "user-data").await
     }
 
-    /// Returns the 'identity document' with fields like region and instance_type.
-    pub async fn fetch_identity_document(&mut self) -> Result<IdentityDocument> {
+    /// Returns the region described in the identity document.
+    pub async fn fetch_region(&mut self) -> Result<Option<String>> {
         let target = "dynamic/instance-identity/document";
-        let response = self.fetch_bytes(target).await?;
-        let identity_document: IdentityDocument =
-            serde_json::from_slice(&response).context(error::Serde)?;
-        Ok(identity_document)
+        let response = match self.fetch_bytes(target).await? {
+            Some(response) => response,
+            None => Vec::new(),
+        };
+        let identity_document: Value = serde_json::from_slice(&response).context(error::Serde)?;
+        let region = identity_document
+            .get("region")
+            .and_then(|value| value.as_str())
```

Comment:
This seems extraneous, but without it the region is returned in quotes, causing the following error when running `pluto cluster-dns-ip`:
```
Unable to parse CIDR from EKS, using default DNS IP: Unable to parse '"us-west-2"' as a region: Not a valid AWS region: "us-west-2"
```

Reply:
I'm a little confused.  You're saying that you need to do the `value.as_str()` bit ?

Reply:
Right. I was just commenting on the fact that it looked a little redundant, but it's necessary.

Reply:
We could eliminate this chain if we deserialize directly into an `IdentityDocument` - see my comment above.

Reply:
`value.as_str()` returns an `Option<&str>` and the following `.map` converts this to an owned `Option<String>`. I could see going back to the IdentityDocument struct but I felt that

`get_region(&mut self) -> Option<String`

was better than

```rust
struct IdentityDocument {
    region: Option<String>
}

fn get_identity_document(&mut self) -> Option<IdentityDocument> { ... }
```

When all the caller wants is the region.

With the struct the caller does

```rust
get_identity_document()
  .unwrap()
  .unwrap()
  .region
  .unwrap()
```

with what we have now, the caller does this which seems better to me:

```rust
get_region()
  .unwrap()
  .unwrap()
```

Reply:
I'm not opposed to using the IdentityDocument as an implementation detail, but it seems more trouble than it's worth given that this is all tested and working as it.